components/user_profile/user_profile_component.js

```diff
@@ -11,14 +11,17 @@ export default class UserProfileComponent extends Component {
     this.state = {
```

Comment:
If this is just displaying a profile, the data you have here should be stored inside of `props`. `state` is for things you need to manage within the component itself, while props is data that is passed down from another component.

So in this case, you should have a 'container' (either called user_profile_container, or just user_profile), and that container will query the server for data about a profile, and then pass it down to this component as props. 

Remember that this component should just be displaying data, it doesn't have any knowledge of how to retrieve the data, just how to display it.